packages/store/src/blob/providers.ts

```diff
@@ -152,11 +221,24 @@ export class BlobCloudSync {
 
       this._taskRunner();
     });
+
+    // resume pending blobs when reconnected
+    this._uploading.keys().then(async keys => {
+      for (const key of keys) {
+        const uploaded = await this._uploading.get(key);
+        if (!uploaded) {
+          this.addTask(key, 'add');
+        }
+      }
+    });
   }
 
   private async _handleTaskRetry(task: SyncTask, status?: BlobStatus) {
+    this._uploadingIds.delete(task.id);
     if (status?.exists) {
       await this._pending.delete(task.id);
+      await this._uploading.set(task.id, true);
```

Comment:
Maybe we can combine the var of padding and uploading? padding uses `type = 'add' | 'delete'` to indicate that it is being added or deleted, and we can add a type of 'updated' to indicate that the upload has been successful

Reply:
Nice I will try only use _pending database .